### PR TITLE
Switch to new VMR control set

### PR DESCRIPTION
**Pull request by mmitche**
Now that roslyn is on 9.0, we can switch to the new control set. Generally:
- DotNetBuildFromSource -> DotNetBuildSourceOnly - Building a source-only build.
- DotnetBuildFromSourceFlavor == Product -> DotNetBuildOrchestrator == true - Building in the VMR, could be source-only or MS's build.
- ArcadeBuildFromSource -> DotNetBuildRepo == true -> Indicates an outer repo build.

### CODE REVIEW

File: eng/build.sh [hunk 1]
```diff
@@ -169,9 +169,8 @@ while [[ $# > 0 ]]; do
     --warnaserror)
       warn_as_error=true
       ;;
-    --sourcebuild|/p:arcadebuildfromsource=true)
-      # Arcade specifies /p:ArcadeBuildFromSource=true instead of --sourceBuild, but that's not developer friendly so we
-      # have an alias.
+    --sourcebuild|/p:dotnetbuildsourceonly=true)
```
Comment (333fred) [inline eng/build.sh:172]: Should we also handle `DotNetBuildRepo=true`?

Comment (mmitche) [reply eng/build.sh:172]: I think it's unlikely that your team will pass that separately. Also, that passing that alone would imply a non-source-only VMR style build. Do you know whether anyone uses that alias?

Comment (333fred) [reply eng/build.sh:172]: I mean, I think we don't use either alias, it's just the arcade template that does :laughing:.

Comment (mmitche) [reply eng/build.sh:172]: Alright I'll investigate further. I didn't think that the VMR actually invokes this script.

Comment (333fred) [reply eng/build.sh:172]: https://github.com/dotnet/roslyn/blob/09ef0c03d033b17f0e5750f265275ed6d1f3cca7/eng/common/core-templates/steps/source-build.yml#L89-L91 was the only place I found that used them.

**Review by 333fred — APPROVED** [review]

**Comment by mmitche** [comment]
Well i'll eat my words now.

File: eng/build.sh (hunk 1 above)
Comment (mmitche) [reply eng/build.sh]: Alright so the crux of why this was necessary is that static graph restore gets disabled under that /p: alias. I moved this to the inner source build arg creation.

**Review by 333fred — APPROVED** [review]

**Pull request merged**